### PR TITLE
Content Identifiers for Software Artifacts

### PR DESCRIPTION
**Pull request by zvr**
This adds a general system for supporting various content identifiers for Software Artifacts.

It adds a new property on `SoftwareArtifact`, its class, and a vocabulary for different types of identifiers (which can be expanded in the future, with no breaking changes).

This is re-work of #611

### CODE REVIEW

File: model/Software/Classes/ContentIdentifier.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+- Instantiability: Concrete
+
+## Properties
+
+- contentIdentifierType
+  - type: ContentIdentifierType
+  - minCount: 1
+  - maxCount: 1
+- contentIdentifierValue
+  - type: xsd:string
```
Comment (rnjudge) [inline model/Software/Classes/ContentIdentifier.md:27]: suggestion to change this to IRI

**Comment by rnjudge** [comment]
Discussion on the tech call 4/9 favored removing the gitoid property in favor of this generalized type.

File: model/Software/Classes/ContentIdentifier.md (hunk 1 above)
Comment (maxhbr) [reply model/Software/Classes/ContentIdentifier.md:27]: As discussed in the tech call:
```suggestion
  - type: xsd:anyURI
```

**Review by maxhbr — APPROVED** [review]
LGTM

File: model/Software/Classes/SoftwareArtifact.md
Comment (JPEWdev) [inline model/Software/Classes/SoftwareArtifact.md:1]: Please remove gitoid as it is redundant and confusing to implementers

File: model/Software/Classes/ContentIdentifier.md (hunk 1 above)
Comment (maxhbr) [reply model/Software/Classes/ContentIdentifier.md:27]: this can be broadened back to string later, if necessary

**Comment by rnjudge** [comment]
@jeff-schutt will review after the call today then we can merge with changes above.

File: model/Software/Properties/contentIdentifierValue.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+## Description
+
+A contentIdentifierValue specifies the value of a content identifier.
+
+## Metadata
+
+- name: contentIdentifierValue
+- Nature: DataProperty
+- Range: xsd:string
```
Comment (zvr) [inline model/Software/Properties/contentIdentifierValue.md:17]:
```suggestion
- Range: xsd:anyURI
```

**Comment by jeff-schutt** [comment]
This PR goes against a prior agreement, documented here:

> Looks good to me. I did verify that we did decide on gitoid as the content identifier in the Nov 1, 2022 tech call.

https://github.com/spdx/spdx-3-model/pull/381#pullrequestreview-1473049459

File: model/Software/Classes/SoftwareArtifact.md
Comment (zvr) [reply model/Software/Classes/SoftwareArtifact.md:1]: @JPEWdev do you want me to do it in this PR, or do you want to approve this one and I can create a new one?

Comment (JPEWdev) [reply model/Software/Classes/SoftwareArtifact.md:1]: A separate PR is fine, but it is a breaking 3.0 change so it needs to be resolved before release

**Comment by zvr** [comment]
It does not "go against" something discussed in 2022, since it still allows using a gitoid as a content identifier.

File: model/Software/Classes/ContentIdentifier.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+- Instantiability: Concrete
+
+## Properties
+
+- contentIdentifierType
+  - type: ContentIdentifierType
+  - minCount: 1
+  - maxCount: 1
+- contentIdentifierValue
+  - type: xsd:anyURI
```
Comment (JPEWdev) [inline model/Software/Classes/ContentIdentifier.md:27]: xsd:string

File: model/Software/Properties/contentIdentifierValue.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+## Description
+
+A contentIdentifierValue specifies the value of a content identifier.
+
+## Metadata
+
+- name: contentIdentifierValue
+- Nature: DataProperty
+- Range: xsd:anyURI
```
Comment (JPEWdev) [inline model/Software/Properties/contentIdentifierValue.md:17]: xsd:string

**Review by JPEWdev — APPROVED** [review]

**Review by kestewart — APPROVED** [review]
This lets a gitoid be considered a content identifier, and lets other standard content identifers that may emerge be incorporated in a structured non breaking way in future.

**Comment by rnjudge** [comment]
Discussed with Kate offline, this PR is ready to merge.

**Pull request merged**